mteb/abstasks/AbsTaskClusteringFast.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import logging
+import random
+from typing import Any, Dict, List
+
+import numpy as np
+import sklearn
+import sklearn.cluster
+from attr import dataclass
+from sklearn.metrics.cluster import v_measure_score
+
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+HFLang = str
+Split = str
+Sentences = List[str]
+ColumnName = str
+ColumnValues = List[str]
+HFDataset = Dict[Split, Dict[ColumnName, ColumnValues]]
+MultilingualDataset = Dict[HFLang, HFDataset]
+IsoLanguage = str
+ClusteringScores = List[Dict[str, float]]
+Scores = Dict[str, Any]
+
+
+def evaluate_clustering_bootstrapped(
+    embeddings: np.ndarray,
+    labels: list[str],
+    n_clusters: int,
+    cluster_size: int,
+    kmean_batch_size: int,
+    rng_state: random.Random = random.Random(),
+) -> ClusteringScores:
+    """Bootstrapped evaluation of clustering performance using V-measure.
+
+    The bootstrapping is done by sampling N samples from the corpus and clustering them. It is done without replacement to get a diverse set of
+    samples.
```

Comment:
With or without replacement. Don't have any strong intuition about what is the best here.

Reply:
As I understand, bootstrapping would mean sampling with replacement. Since we have downsampled at the encode step, bootstrapping should help "raise" the number of samples same as the original size. I'm not sure if we can achieve that if we sampled without (like in train_test_split). We could use `choices` instead:
```
random.choices(seq, k)
```

Reply:
Yep, the problem I found was that it inflated the score slightly (due to the same embeddings would naturally end up in the same cluster). I am not sure it is that big of a problem though

Reply:
I went with the classic bootstrapping approach you suggested